### DISCUSSION TITLE
How to replace instead of insert?

### DISCUSSION THREAD

**Discussion by newalphamedia** [ueberdosis/tiptap#1980] [General Questions] [answered]
My tiptap editor uses both HardBreak and Paragraph:
**import HardBreak from '@tiptap/extension-hard-break'
import Paragraph from '@tiptap/extension-paragraph'**

Currently, I use the following to insert a line at the current position:
**this.editor.chain().focus().insertContent(line).run();**

The issiue is, that if the editor content is empty, then it will look like this after I insertContent:
<p></p>
<p>Inserted Line</p>

However, what I want is for the editor content to look like this:
<p>Inserted Line</p>

We could check if **this.editor.state.selection.$anchor.nodeBefore** is null - and then do a replaceContent but I can't find any such function or similar methods by looking at the prosemirror docs.

Any ideas?

**Comment by philippkuehn** [comment, marked as answer]
You can create a custom `command`, calculate a `range` and call `insertContentAt` to replace a range.

```js
editor.commands.command(({ tr, commands }) => {
  // get current selection
  const selection = tr.selection

  // calculate range to replace
  const range = {
    from: // ...
    to: // ...
  }
  const content = // ... your content
  
  return commands.insertContentAt(range, content)
})
```

**Comment by newalphamedia** [comment]
Thank you, I was able to find a solution with your response.
